### DISCUSSION TITLE
There is a problem with the share file url

### DISCUSSION THREAD

**Discussion by pjy324** [minio/minio#14695] [Q&A]
## Expected Behavior
share url shall be externally available in Kubernetes environment

## Current Behavior
In the Kubernetes environment, share url is transferred to the internal ip or internal domain.
Is there any way or option to resolve this issue?

## Steps to Reproduce (for bugs)
1. Case 1

[root@node2 standalon]# kubectl get all -n jin-minio-standalone -o wide
NAME                                    READY   STATUS    RESTARTS   AGE   IP             NODE    NOMINATED NODE   READINESS GATES
pod/minio-deployment-75f868bbb6-jgggd   1/1     Running   0          10s   10.233.95.66   node7   <none>           <none>

If you press "Share File" through the web, the following url appears. This URL is not accessible from outside
http://10.233.95.66:19000~~~~

2. Case2
[root@node1 ~]# kubectl get all -n dl-dlm-jin-test -o wide
NAME                           READY   STATUS    RESTARTS   AGE   IP              NODE    NOMINATED NODE   READINESS GATES
pod/dl-dlm-jin-test-pool-0-0   1/1     Running   0          11h   10.233.108.23   node6   <none>           <none>
pod/dl-dlm-jin-test-pool-0-1   1/1     Running   0          11h   10.233.95.25    node7   <none>           <none>
pod/dl-dlm-jin-test-pool-0-2   1/1     Running   0          11h   10.233.70.24    node5   <none>           <none>
pod/dl-dlm-jin-test-pool-0-3   1/1     Running   0          11h   10.233.105.21   node4   <none>           <none>

https://minio.dl-dlm-jin-test.svc.cluster.local~~~

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by klauspost** [comment]
Have you set https://docs.min.io/minio/baremetal/reference/minio-server/minio-server.html?#envvar.MINIO_SERVER_URL  - set it to the URL of your load balancer.

Use that with the client that generates the URL.
